controllers/edgedeployment_controller.go

```diff
@@ -236,13 +259,47 @@ func (r *EdgeDeploymentReconciler) getLabelledEdgeDevices(ctx context.Context, n
 	return r.EdgeDeviceRepository.ListForSelector(ctx, &selector, namespace)
 }
 
+func (r *EdgeDeploymentReconciler) executeConcurrent(ctx context.Context, f ConcurrentFunc, input []managementv1alpha1.EdgeDevice) []error {
+	var errs []error
+	if r.Concurrency == 1 {
+		errs = f(input)
+	} else {
+		errs = r.ConcurrentExecutor(r.Concurrency, f, input)
+	}
+	return errs
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *EdgeDeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&managementv1alpha1.EdgeDeployment{}).
 		Complete(r)
 }
 
+func ConcurrentExecutor(concurrency uint, f ConcurrentFunc, input []managementv1alpha1.EdgeDevice) []error {
```

Comment:
Please change `input` to something meaningful, like `edgeDevices`.

Reply:
I will change input to edgeDevices in order to save the debate. Although, I do not agree that it makes a difference or that it is wrong..
I did not understand what is wrong with the name 'ConcurrentExecutor'. The name describes exactly what the function is supposed to do.

Reply:
It's a noun; function by definition is not representing some object/thing (noun), but action (verb). I'd accept a noun name of a function for an accessor or a constructor. It's neither.

Reply:
That is very much arguable and I do not agree with this approach.

Reply:
In general there are naming conventions which should be followed. Here are [golang](https://go.dev/doc/effective_go#names) naming conventions. We can come up with our own if we want to adjust generic golang conventions. It is important to have consistent code in the project.

Reply:
I did not find something in golang's conventions that contradicts the name I used. I use golint and other go tools that help me follow the conventions.
I think that the debate here is not about conventions but personal flavor.

Reply:
Let's talk about this. As far as I can see nothing personal here.

Reply:
@jakub-dzon I will change it to ExecuteConcurrent. Please add explanations to contributing guidelines so it will be easier to solve these issues in the future.